openedx/frontend-app-discussions
# feat: Add support for loading themes at runtime

### Description

This PR updates this repo to use the version of Paragon with design tokens, and runtime theming. 

#### How Has This Been Tested?

To test the changes in this PR you need to set things up as follows:

1. Run `npm install --legacy-peer-deps` to install the packages. This is installing current WIP versions so this allows installation despite conflicts. 
2. Create `module.config.js` file with the following contents: 
```javascript
module.exports = {
  localModules: [
    { moduleName: '@edx/frontend-platform', dir: '../frontend-platform', dist: 'src' },
  ],
};
```
Note that the 'dir' key above should point to where you've checked out `frontend-platform` to the correct version. i.e the version in pacakge.json
3. If using tutor, shut down the discussion MFE and add the following content to .env.private:
```bash
BASE_URL='http://apps.local.overhang.io:2002'
CREDENTIALS_BASE_URL='http://localhost:18150'
ECOMMERCE_BASE_URL='http://localhost:18130'
LMS_BASE_URL='http://local.overhang.io:8000'
LEARNING_BASE_URL='http://apps.local.overhang.io:2000'
LOGIN_URL='http://local.overhang.io:8000/login'
LOGOUT_URL='http://local.overhang.io:8000/logout'
MARKETING_SITE_BASE_URL='http://local.overhang.io:8000'
ORDER_HISTORY_URL='http://localhost:1996/orders'
REFRESH_ACCESS_TOKEN_ENDPOINT='http://local.overhang.io:8000/login_refresh'
APP_ID='discussions'
MFE_CONFIG_API_URL='http://local.overhang.io:8000/api/mfe_config/v1'
```
And add the following to webpack.dev.config.js:
```javascript
const { createConfig } = require('@edx/frontend-build');
const config = createConfig('webpack-dev');
config.devServer.allowedHosts = ['apps.local.overhang.io'];
module.exports = config;
```
This will allow loading the MFE from apps.local.overhang.io:2002. 


#### Screenshots/sandbox (optional):
Include a link to the sandbox for design changes or screenshot for before and after. **Remove this section if it's not applicable.**

|Before|After|
|-------|-----|
|      |      |

#### Merge Checklist


#### Post-merge Checklist
